File: 10828.py
"""
파일 이름 : 10828.py
제작자 : 정지운
제작 날짜 : 2017년 8월 10일
제작 목표 : 스택을 구현하고 활용한다.
"""

""" 런타임 에러 """

stk = []

# 입력
tc = int(input())

for i in range(tc):
	# 입력
	s = input()

	# 명령 실행
	if s == "pop":
		try:
			print(stk.pop())
		except IndexError:
			print(-1)
	elif s == "size":
		print(len(stk))
	elif s == "empty":
		if len(stk) == 0:
			print(1)
		else:
			print(0)
	elif s == "top":
		print(stk[len(stk) - 1])
	else:
		num = int(s.split(' ')[1])
		stk.append(num)

File: 10828.cpp
#include <iostream>
#include <stack>
#include <string>
using namespace std;

int n;
stack<int> s;

void GetInput()
{
    cin.tie(nullptr);
    ios::sync_with_stdio(false);

    cin >> n;
}

void Solve()
{
    for (int i = 0; i < n; i++)
    {
        string str;
        cin >> str;

        if (str == "pop")
        {
            if (!s.empty())
            {
                cout << s.top() << '\n';
                s.pop();
            }
            else
                cout << -1 << '\n';
        }
        else if (str == "size")
        {
            cout << s.size() << '\n';
        }
        else if (str == "empty")
        {
            if (s.empty())
                cout << 1 << '\n';
            else
                cout << 0 << '\n';
        }
        else if (str == "top")
        {
            if (!s.empty())
                cout << s.top() << '\n';
            else
                cout << -1 << '\n';
        }
        else
        {
            int num;
            cin >> num;
            s.push(num);
        }
    }
}

int main(void)
{
    GetInput();
    Solve();
    return 0;
}
